internal/observability.go

```diff
@@ -28,80 +28,86 @@ import (
 	"go.opencensus.io/stats/view"
 	"go.opencensus.io/tag"
 	"go.opencensus.io/trace"
-
-	commonpb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1"
-	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
 )
 
 var (
-	tagKeyReceiverName, _ = tag.NewKey("opencensus_receiver")
-	tagKeyExporterName, _ = tag.NewKey("opencensus_exporter")
-)
-
-var mReceivedSpans = stats.Int64("oc.io/receiver/received_spans", "Counts the number of spans received by the receiver", "1")
+	tagKeyTraceReceiverName, _ = tag.NewKey("opencensus_trace_receiver")
+	mReceiverReceivedSpans     = stats.Int64("oc.io/receiver/received_spans", "Counts the number of spans received by the receiver", "1")
+	mReceiverDroppedSpans      = stats.Int64("oc.io/receiver/dropped_spans", "Counts the number of spans dropped by the receiver", "1")
 
-var itemsDistribution = view.Distribution(
-	0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 12, 14, 16, 18, 20, 25, 30, 35, 40, 45, 50, 60, 70, 80, 90,
-	100, 150, 200, 250, 300, 450, 500, 600, 700, 800, 900, 1000, 1200, 1400, 1600, 1800, 2000,
+	tagKeyTraceExporterName, _ = tag.NewKey("opencensus_trace_exporter")
+	mExporterReceivedSpans     = stats.Int64("oc.io/exporter/received_spans", "Counts the number of spans received by the exporter", "1")
+	mExporterDroppedSpans      = stats.Int64("oc.io/exporter/dropped_spans", "Counts the number of spans received by the exporter", "1")
 )
 
-// ViewReceivedSpansReceiver defines the view for the received spans metric.
-var ViewReceivedSpansReceiver = &view.View{
-	Name:        "oc.io/receiver/received_spans",
-	Description: "The number of spans received by the receiver",
-	Measure:     mReceivedSpans,
-	Aggregation: itemsDistribution,
-	TagKeys:     []tag.Key{tagKeyReceiverName},
+// ViewReceiverReceivedSpans defines the view for the receiver received spans metric.
+var ViewReceiverReceivedSpans = &view.View{
+	Name:        mReceiverReceivedSpans.Name(),
+	Description: mReceiverReceivedSpans.Description(),
+	Measure:     mReceiverReceivedSpans,
+	Aggregation: view.Sum(),
+	TagKeys:     []tag.Key{tagKeyTraceReceiverName},
 }
 
-var mExportedSpans = stats.Int64("oc.io/receiver/exported_spans", "Counts the number of exported spans", "1")
+// ViewReceiverDroppedSpans defines the view for the receiver dropped spans metric.
+var ViewReceiverDroppedSpans = &view.View{
+	Name:        mReceiverDroppedSpans.Name(),
+	Description: mReceiverDroppedSpans.Description(),
+	Measure:     mReceiverDroppedSpans,
+	Aggregation: view.Sum(),
+	TagKeys:     []tag.Key{tagKeyTraceReceiverName},
+}
+
+// ViewExporterReceivedSpans defines the view for the exporter received spans metric.
+var ViewExporterReceivedSpans = &view.View{
+	Name:        mExporterReceivedSpans.Name(),
+	Description: mExporterReceivedSpans.Description(),
+	Measure:     mExporterReceivedSpans,
+	Aggregation: view.Sum(),
+	TagKeys:     []tag.Key{tagKeyTraceReceiverName, tagKeyTraceExporterName},
```

Comment:
why tagKeyTraceReceiverName is here?

Reply:
That is the source so you want all these metrics to be broken down by [receiver, exporter].

Reply:
To find more details you can read about "tags". Short story we add the "oc_receiver" tag to the context so all the stats we record against that context will have this tag available, it is exactly what you did by manually propagating the "source" in the internal/collector/processor.

Reply:
I see so if the context flowing correctly we will know the source, nice.